### PR TITLE
Fix: Serialize TimeOfDay

### PR DESCRIPTION
**Pull request by FrankEssenberger**
## Context

We had a bug in the serializer for single digit values in the `Time` object. The padding with zeros was missing.
So `{hours:1,minutes:2,seconds:3}` was serialized to `1:2:3` and not `01:02:03` as it should be.

## Definition of Done

Please consider all items and remove only if not applicable.

- [x] Tests created/adjusted for your changes.
- [x] Release notes updated.
  * Provide sufficient context so that each entry can be understood on its own.
  * Be specific about names of functions, classes, modules, etc.
  * Describe when or where this is relevant
  * Use indicative and present tense. For example, write "Provide function `name` that does X in order to Y" over "Now X can be done by calling a new function".
- [x] PR title adheres to [conventional commit guidelines](https://www.conventionalcommits.org) (please note that only `fix:` and `feat:` will end up in the release notes)
~~- [ ] If applicable: Properly documented (JSDoc of public API)~~
~~- [ ] If applicable: Check if `node run doc` still works.~~

### CODE REVIEW

File: packages/core/src/odata/v4/payload-value-converter.ts [hunk 1]
```diff
@@ -107,9 +107,22 @@ function durationToEdmDuration(value: moment.Duration): string {
...
+function padSeconds(seconds: number): string {
+  const secondsString = seconds.toString(10);
+  if (!secondsString.includes('.')) {
+    return secondsString.padStart(2, '0');
+  }
+  const secondsStringParts: string[] = secondsString.split('.');
+  return [secondsStringParts[0].padStart(2, '0'), secondsStringParts[1]].join(
+    '.'
+  );
+}
```
Comment (marikaner) [inline packages/core/src/odata/v4/payload-value-converter.ts:125]: Actually you could use this function for all parts of the timeofday but even if you don't like that take a look below:
```suggestion
function padTimeComponent(timeComponent: number): string {
  const [wholeNumber, fractionalNumber] = timeComponent.toString().split('.');
  return [wholeNumber.padStart(2, '0'), fractionalNumber].join('.');
}
```

File: packages/core/src/odata/v4/payload-value-converter.ts [hunk 2]
```diff
@@ -107,9 +107,22 @@ function durationToEdmDuration(value: moment.Duration): string {
 }
 
 function timeToEdmTimeOfDay(value: Time): string {
-  return `${value.hours}:${value.minutes}:${value.seconds}`;
+  const hoursPadded = value.hours.toString(10).padStart(2, '0');
+  const minutesPadded = value.minutes.toString(10).padStart(2, '0');
+  const secondsPadded = padSeconds(value.seconds);
+  return `${hoursPadded}:${minutesPadded}:${secondsPadded}`;
```
Comment (marikaner) [inline packages/core/src/odata/v4/payload-value-converter.ts:113]: I think you can repeat yourself a little less, also see suggestion below.
```suggestion
  return [value.hours, value.minutes, value.seconds].map(timeComponent => padTimeComponent(timeComponent)).join(':');
```

**Review by marikaner — CHANGES_REQUESTED** [review]
Minor changes, let me know what you think!

**Comment by FrankEssenberger** [comment]
liked your suggestions

**Review by marikaner — APPROVED** [review]

**Comment by FrankEssenberger** [comment]
Very good remarks marika.

**Pull request merged**